src/components/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,154 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable react/jsx-props-no-spreading */
+import Box from "@mui/material/Box";
+import ClickAwayListener from "@mui/material/ClickAwayListener";
+import Divider from "@mui/material/Divider";
+import Grow from "@mui/material/Grow";
+import MenuList from "@mui/material/MenuList";
+import Paper from "@mui/material/Paper";
+import Popper, { PopperPlacementType, PopperProps } from "@mui/material/Popper";
+import { useTheme } from "@mui/material/styles";
+import useMediaQuery from "@mui/material/useMediaQuery";
+import React, { useCallback, useRef, useState } from "react";
+
+import { DefaultDropdownButton } from "./DefaultDropdownButton";
+import { DropdownItem } from "./DropdownItem";
+
+export interface DropdownAction {
+    id: string;
+    disabled: boolean;
+    content: string;
+    icon: JSX.Element;
+    shouldMenuStayOpened?: boolean;
+    key?: string;
+    showCheckIcon?: boolean;
+    isShown?: boolean;
+    isSelected?: boolean;
+    isDivider?: boolean;
+    onClick: (action: DropdownAction) => void;
+}
+
+export interface DropdownProps {
+    id?: string;
+    popperProps?: {
+        id: string;
+        modifiers?: PopperProps["modifiers"];
+        "data-popper-id"?: string;
+    };
+    buttonContent?: string;
+    actions: DropdownAction[];
+    children?: React.ReactNode | React.ReactNode[];
+    paperPlacement?: PopperPlacementType;
+    className?: string;
+    disabled?: boolean;
+}
+
+/**
+ *  MUI dropdown component have a default button with dropdown also could be used with
+ * custom button which takes from children, actions array -> array which will be converted
+ * to dropdown menu items.
```

Comment:
This sentence is not understandable, plus has many grammatic errors

Reply:
Sure it is. I've noticed it too,